core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -819,7 +827,7 @@ private Builder(TableMetadata base) {
       this.startingChangeCount = changes.size();
 
       this.snapshotLog = Lists.newArrayList(base.snapshotLog);
-      this.previousFileLocation = base.metadataFileLocation;
+      this.previousFileLocation = noPreviousFile ? null : base.metadataFileLocation;
```

Comment:
a little bit annoying that this needs to be set, so I wonder whether it would be better to have a builder method that allows overriding this

Reply:
I need to override that too for something I'm working on. But I have just used the builder's `withMetadataLocation`. In my case, the value of `base.metadataFIleLocation` is already null so just calling the function works fine.